src/charm.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+import subprocess
+from urllib.request import urlopen
+
+from ops.charm import CharmBase
+from ops.main import main
+from ops.model import BlockedStatus, MaintenanceStatus, WaitingStatus
+from charms.operator_libs_linux.v0 import apt
+
+logger = logging.getLogger(__name__)
+
+
+class MongodbOperatorCharm(CharmBase):
+    """Charm the service."""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.framework.observe(self.on.install, self._on_install)
+
+    def _on_install(self, _):
+        """Install prerequisites for the application"""
+        self.unit.status = MaintenanceStatus("installing MongoDB")
+        self._add_mongodb_org_repository()
+        self._install_apt_packages(["mongodb-org"])
+        self.unit.status = WaitingStatus("Waiting to start MongoDB")
+
+    def _add_mongodb_org_repository(self):
+        repositories = apt.RepositoryMapping()
+
+        # Get GPG key
+        try:
+            key = urlopen("https://www.mongodb.org/static/pgp/server-5.0.asc").read().decode()
+        except Exception as e:
+            logger.exception(f"failed to get GPG key", exc_info=e)
+            self.unit.status = BlockedStatus("Failed to get GPG key")
+            return
+
+        # Add the repository if it doesn't already exist
+        if "https://repo.mongodb.org/apt/ubuntu" not in repositories:
```

Comment:
Oh just one thing -- check that this is the right format... should this not be: `repo-mongodb-org-apt-ubuntu` or similar? You'll need to check what's in the repository mapping to be sure :)